feat: 주문 도메인 기능 구현

wander0000 opened the pull request:
작업내용 - 주문도메인 기능 구현

- 주문 Dto 추가
- 주문 등록 기능 구현
- 주문 취소 기능 구현
- 주문 상세 조회 기능 구현
- 주문 조회(페이징, 필터링-매장별, 사용자별, 정렬-생성일,수정일,매장이름,사용자이름) 기능 구현

wander0000 committed:
feat: 주문 Dto 추가

wander0000 committed:
feat: 주문 등록 기능 구현

wander0000 committed:
feat: 주문 취소 기능 구현

wander0000 committed:
fix: 주문 응답 Dto 수정 및 추가

wander0000 committed:
fix: 주문 생성 기능 수정(응답Dto 변경)

wander0000 committed:
feat: 주문 조회(페이징,필터링) 기능 구현

wander0000 committed:
feat: 주문 상세조회 기능 구현

wander0000 committed:
feat: 주문 수정 기능 구현

wander0000 committed:
chore: 미사용메소드 삭제, 코드정리

yuhyun1 reviewed (approved):
👍

olrlobt reviewed (approved):
🚀

jshstar reviewed (approved):
👍  추가적으로 구조 리펙토링 진행했습니다.

jshstar committed:
fix: merge conflict 해결

jshstar committed:
refactor: 주문 생성 로직 리펙토링
OrderProduct 엔티티 생성에서 서비스 생성으로 변경
OrderInfoDto 에서 OrderInfoResponse 명으로 변경

jshstar committed:
fix: 조회 파라미터 변경

jshstar committed:
fix: merge conflict 누락 부분 수정

jshstar committed:
Merge branch 'main' into feat/order

jshstar committed:
feat: PageableArgumentResolver 기능 구현
PageAble의 size 범위의 제한을 위해 사용

jshstar committed:
feat: pageableConfig 구현

jshstar committed:
chore: 필드명 변경

jshstar committed:
refactor: 해당 매장의 주문 검색 조회 및 고객 주문 조회 기능 리펙토링
유저와 관리자의 입장에서 입력받을 수 있는 파라미터의 조건을 고려해서 구현 기존에 order와 orderProduct를 함께 반환하는 구조에서 order만 반환하는 구조로 변경

jshstar committed:
chore: 필드명 변경

jshstar committed:
chore: 클래스명 변경

jshstar committed:
refactor: 주문 세부정보 조회 리펙토링
order와 orderProuct를 분리했기 때문에 orderId에 대한 orderProduct 조회를 진행 또한 고객만 조회시 실제로 고객이 주문한 것인지 검증

jshstar committed:
refactor: 주문 업데이트 로직 리펙토링
주문 요청사항, 주소 수정등 수정 가능 범위 확장 및 주소 검증 로직 추가

jshstar committed:
refactor: 주문 업데이트 로직 리펙토링
주문 요청사항, 주소 수정등 수정 가능 범위 확장 및 주소 검증 로직 추가

jshstar committed:
refactor: 주문 삭제 기능 리펙토링
시간 검증 부분 메서드로 분리

jshstar committed:
feat: todo 검증 처리 추가 구현

jshstar committed:
fix: url 변경

jshstar committed:
chore: 불필요한 메서드 정리

jshstar committed:
chore: 불필요한 주입 정리

jshstar committed:
refactor: ModelAttribute를 사용하여 여러 파라미터를 Dto로 통합 및 검증 리팩토링
- 주문 관리 요청에서 사용되는 파라미터 구조를 일관성 있게 정리
- 코드 가독성 및 유지보수성 향상

jshstar committed:
refactor: 생성자를 활용해 각 리스폰스 초기화 리팩토링
- 파라미터 리스트를 간결하게 정리하여 코드 가독성 향상
- 중복된 데이터 초기화 로직 제거 및 유지보수성 개선
